src/Tests/dotnet-workload-install.Tests/WorkloadGarbageCollectionTests.cs

```diff
@@ -0,0 +1,393 @@
+            CreateMockManifest("testmanifest", "3.0.0", "6.0.300", sourceManifestName: @"Sample2_v3.json");
+
+            CreateManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300");
+            CreateManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300");
+            CreateManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300");
+
+            CreateInstallState("6.0.300",
+               """
+                {
+                    "manifests": {
+                        "testmanifest": "2.0.0/6.0.300",
+                    }
+                }
+                """);
+
+            var (installer, getResolver) = GetTestInstaller("6.0.300");
+
+            var packsToKeep = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk),
+                CreatePackInfo("Xamarin.Android.Framework", "8.5.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.5.0.1", WorkloadPackKind.Library)
+            };
+
+            var packsToCollect = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Framework", "8.6.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.6.0.0", WorkloadPackKind.Library)
+            };
+
+            foreach (var pack in packsToKeep.Concat(packsToCollect))
+            {
+                CreateInstalledPack(pack, "6.0.300");
+            }
+
+            // Write workload install record for 6.0.300
+            var workloadsRecordPath = Path.Combine(_dotnetRoot, "metadata", "workloads", "6.0.300", "InstalledWorkloads");
+            Directory.CreateDirectory(workloadsRecordPath);
+            File.Create(Path.Combine(workloadsRecordPath, "xamarin-android-build"));
+
+            installer.GarbageCollect(getResolver);
+
+            foreach (var pack in packsToCollect)
+            {
+                PackShouldExist(pack, false);
+                PackRecord(pack, "6.0.300").Should().NotExist();
+            }
+            foreach (var pack in packsToKeep)
+            {
+                PackShouldExist(pack, true);
+                PackRecord(pack, "6.0.300").Should().Exist();
+            }
+
+            ManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300").Should().NotExist();
+            ManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300").Should().Exist();
+            ManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300").Should().NotExist();
+
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.100", "testmanifest", "1.0.0", "WorkloadManifest.json")).Should().NotExist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "2.0.0", "WorkloadManifest.json")).Should().Exist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "3.0.0", "WorkloadManifest.json")).Should().NotExist();
```

Comment:
Should we also check that the sdk-manifests folder for 6.0.100 no longer exists?

Reply:
I tried this out but we don't actually walk up the sdk-manifests tree and delete empty folders like we do with the installation records.  Normally I would probably fix that but at this point I want to keep the PR restricted to test changes.

Reply:
I made a remark about this here: https://github.com/dotnet/sdk/issues/35455. It would definitely be easy to write an unsuspecting bug that expects the folders to be deleted, so eventually it should be fixed, though I agree for now it's OK